salt/states/ssh_auth.py

```diff
@@ -454,35 +435,26 @@ def absent(
         )
         return ret
 
-    # Extract Key from file if source is present
+    # Get only the path to the file without env referrences to check if exists
     if source != "":
-        key = __salt__["cp.get_file_str"](source, saltenv=__env__)
-        filehasoptions = False
-        # check if this is of form {options} {enc} {key} {comment}
-        sshre = re.compile(r"^(sk-)?(ssh\-|ecds).*")
-        key = key.rstrip().split("\n")
-        for keyline in key:
-            filehasoptions = sshre.match(keyline)
-            if not filehasoptions:
-                ret["comment"] = __salt__["ssh.rm_auth_key_from_file"](
-                    user,
-                    source,
-                    config,
-                    saltenv=__env__,
-                    fingerprint_hash_type=fingerprint_hash_type,
-                )
-            else:
-                # Split keyline to get key
-                keyline = keyline.split(" ")
-                ret["comment"] = __salt__["ssh.rm_auth_key"](
-                    user,
-                    keyline[1],
-                    config=config,
-                    fingerprint_hash_type=fingerprint_hash_type,
-                )
+        source_path = __salt__["cp.get_url"](source, None, saltenv=__env__)
+
+    # Extract Key from file if source is present
+    if source != "" and not source_path:
+        data = "no key"
+    elif source != "" and source_path:
```

Comment:
We already checked `source != ""` on line 439. Maybe just move these into that block:
```python
    if source != "":
        source_path = __salt__["cp.get_url"](source, None, saltenv=__env__)

        # Extract Key from file if source is present
        if not source_path:
            data = "no key"
        else:
            ...
```

Reply:
Oh, I only copy the same comments and code snippet from `ssh_auth.present`. I will change this code here and check if it's Ok be changed there.

Reply:
Just changed the two instances of this pattern (in `ssh_auth.absent` and `ssh_auth.present`)